civis/io/_tables.py

```diff
@@ -1027,10 +1027,23 @@ def civis_file_to_table(file_id, database, table, client=None,
     except ValueError:
         table_exists = False
 
+    table_columns_valid = True
+    if table_columns:
+        table_columns_valid = True if table_columns[0]['sql_type'] else False
+        for i in range(0, len(table_columns)-1):
+            thisSqlType = table_columns[i]['sql_type']
+            nextSqlType = table_columns[i+1]['sql_type']
+            if not thisSqlType == nextSqlType:
```

Comment:
Yeah this might be confusing to read, but actually putting the `not` before a string evaluates it as a boolean. So what I am checking here is that if the next sql type exists but the previous one does not, or vice versa, then we throw an error. Because that means that there is a mix between existing and non-existing sql types. If the error isn't thrown then the original assignment of True or False remains, the former means they are all provided and the latter means none are provided. I'm pretty sure this logic achieves what we wants and I like that it exits the loop as early as necessary. But let me use a test just to be sure!

Reply:
I think in Python we can arrive at a much simpler approach that is easier to read. Python has a lot of powerful built-ins.

For example - `sql_type_cnt = sum(1 for col in table_columns if col.get('sql_type'))`

Reply:
Yeah that is simpler, I'll go with this approach.

Reply:
Done. That was easy!
I haven't used Python in a really long time, so it's hard for me to pull out these tools.